src/traverse.js

```diff
@@ -0,0 +1,34 @@
+import { base58btc } from 'multiformats/bases/base58'
+
+/**
+ * @typedef {import('./cid.js').CID} CID
+ */
+
+/**
+ * @template T
+ * @typedef {import('./block.js').Block<T>} Block
+ */
+
+/**
+ * @template T
+ * @param {Object} options
+ * @param {CID} options.cid
+ * @param {(cid: CID) => Promise<Block<T>>} options.load
+ * @param {Set<string>} options.seen
+ */
+const walk = async ({ cid, load, seen = new Set() }) => {
+  const b58Cid = cid.toString(base58btc)
+  if (seen.has(b58Cid)) {
+    return
+  }
+
+  const block = await load(cid)
+  seen.add(b58Cid)
+
+  for (const link of block.links()) {
+    const linkCid = link[1]
+    await walk({ cid: linkCid, load, seen })
+  }
```

Comment:
```suggestion
  for (const [_, cid] of block.links()) {
    await walk({ cid, load, seen })
  }
```

I think this would work?